### PR TITLE
refactor: used Call.Factory instead of OkHttpClient while generating client implementation

### PR DESCRIPTION
**Pull request by vicmosin**
See https://github.com/OpenAPITools/openapi-generator/pull/19422 for explanation

### CODE REVIEW

**Review by MaSch0212 — CHANGES_REQUESTED** [review]

File: packages/kotlin/assets/client/okhttp3/ApiClient.kt [hunk 1]
```diff
@@ -22,7 +22,7 @@ import java.time.OffsetDateTime
 import java.time.OffsetTime
 import java.util.Locale
 
-open class ApiClient(val baseUrl: String, val client: OkHttpClient = defaultClient) {
+open class ApiClient(val baseUrl: String, val client: Call.Factory = defaultClient) {
```
[the remark this answers is not in the document]
Comment (vicmosin) [reply packages/kotlin/assets/client/okhttp3/ApiClient.kt]: Done

File: packages/kotlin/src/ast/references/okhttp3.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { ktReference, type KtReferenceFactory } from '../nodes/reference.ts';
 
 // okhttp3
-export const okHttpClient: KtReferenceFactory = ktReference.factory('OkHttpClient', 'okhttp3');
+export const okHttpClient: KtReferenceFactory = ktReference.factory('Call.Factory', 'okhttp3');
```
[the remark this answers is not in the document]
Comment (vicmosin) [reply packages/kotlin/src/ast/references/okhttp3.ts]: Sone

File: packages/kotlin/src/generators/services/okhttp3-clients/okhttp3-client-generator.ts [hunk 3]
```diff
@@ -67,7 +67,9 @@ export class DefaultKotlinOkHttp3Generator extends KotlinFileGenerator<Context,
       primaryConstructor: kt.constructor(
         [
           kt.parameter('basePath', kt.refs.string(), { default: 'defaultBasePath' }),
-          kt.parameter('client', kt.refs.okhttp3.okHttpClient(), { default: kt.call(['ApiClient', 'defaultClient']) }),
+          kt.parameter('client', kt.refs.okhttp3.okHttpClient(), {
+            default: kt.call(['defaultClient']),
```
Comment (MaSch0212) [inline packages/kotlin/src/generators/services/okhttp3-clients/okhttp3-client-generator.ts:71]: `kt.call` is unnecessary here. You can just write `default: 'defaultClient'`

**Review by MaSch0212 — APPROVED** [review]

File: packages/kotlin/src/generators/services/okhttp3-clients/okhttp3-client-generator.ts (hunk 3 above)
Comment (vicmosin) [reply packages/kotlin/src/generators/services/okhttp3-clients/okhttp3-client-generator.ts]: Done

**Pull request merged**